hi-ml-histopathology/src/histopathology/utils/report_utils.py

```diff
@@ -0,0 +1,339 @@
+
+
+def collect_crossval_metrics(parent_run_id: str, download_dir: Path, aml_workspace: Workspace,
+                             crossval_arg_name: str = "cross_validation_split_index") -> pd.DataFrame:
+    """Fetch metrics logged to Azure ML from cross-validation runs as a dataframe.
+
+    Will only download the metrics if they do not already exist locally, as this can take several
+    seconds for each child run.
+
+    :param parent_run_id: Azure ML run ID for the parent Hyperdrive run.
+    :param download_dir: Directory where to save the downloaded metrics as `aml_metrics.pickle`.
+    :param aml_workspace: Azure ML workspace in which the runs were executed.
+    :param crossval_arg_name: Name of the Hyperdrive argument used for indexing the child runs.
+    :return: A dataframe in the format returned by :py:func:`~health_azure.aggregate_hyperdrive_metrics()`.
+    """
+    # Save metrics as a pickle because complex dataframe structure is lost in CSV
+    metrics_pickle = download_dir / "aml_metrics.pickle"
+    if metrics_pickle.is_file():
+        print(f"AML metrics file already exists at {metrics_pickle}")
+        with open(metrics_pickle, 'rb') as f:
+            metrics_df = pickle.load(f)
+    else:
+        metrics_df = aggregate_hyperdrive_metrics(run_id=parent_run_id,
+                                                  child_run_arg_name=crossval_arg_name,
+                                                  aml_workspace=aml_workspace)
+        metrics_pickle.parent.mkdir(parents=True, exist_ok=True)
+        print(f"Writing AML metrics file to {metrics_pickle}")
+        with open(metrics_pickle, 'wb') as f:
+            pickle.dump(metrics_df, f)
+    return metrics_df.sort_index(axis='columns')
+
+
+def plot_roc_curve(labels: Sequence, scores: Sequence, label: str, ax: Axes) -> None:
+    """Plot ROC curve for the given labels and scores, with AUROC in the line legend.
+
+    :param labels: The true binary labels.
+    :param scores: Scores predicted by the model.
+    :param label: An line identifier to be displayed in the legend.
+    :param ax: `Axes` object onto which to plot.
+    """
+    fpr, tpr, _ = roc_curve(labels, scores)
+    auroc = auc(fpr, tpr)
+    label = f"{label} (AUROC: {auroc:.3f})"
+    ax.plot(fpr, tpr, label=label)
+
+
+def plot_pr_curve(labels: Sequence, scores: Sequence, label: str, ax: Axes) -> None:
+    """Plot precision-recall curve for the given labels and scores, with AUROC in the line legend.
+
+    :param labels: The true binary labels.
+    :param scores: Scores predicted by the model.
+    :param label: An line identifier to be displayed in the legend.
+    :param ax: `Axes` object onto which to plot.
+    """
+    precision, recall, _ = precision_recall_curve(labels, scores)
+    aupr = auc(recall, precision)
+    label = f"{label} (AUPR: {aupr:.3f})"
+    ax.plot(recall, precision, label=label)
+
+
+def format_pr_or_roc_axes(plot_type: str, ax: Axes) -> None:
```

Comment:
This function could be more generic - `format_plot_axes(axs: Axes, xlabel: str, ylabel:str, xlim_min: float, xlim_max: float, ...)`

Reply:
Here I wanted to just standardise the look of PR/ROC plots. I'm not sure there's much value in making it generic, as the properties can easily be modified on their own.